zkstack_cli/crates/zkstack/src/commands/dev/commands/fmt.rs

```diff
@@ -2,124 +2,69 @@ use std::path::PathBuf;
     Ok(Cmd::new(prettier_command).run()?)
 }
 
-async fn rustfmt(shell: Shell, check: bool, link_to_code: PathBuf) -> anyhow::Result<()> {
-    for dir in ["core", "prover", "zkstack_cli"] {
-        let spinner = Spinner::new(&msg_running_rustfmt_for_dir_spinner(dir));
-        let _dir = shell.push_dir(link_to_code.join(dir));
-        let mut cmd = cmd!(shell, "cargo fmt -- --config imports_granularity=Crate --config group_imports=StdExternalCrate");
-        if check {
-            cmd = cmd.arg("--check");
-        }
-        spinner.freeze();
-        Cmd::new(cmd).run()?;
-    }
-    Ok(())
-}
-
-async fn run_all_rust_formatters(
+async fn rustfmt(
     shell: Shell,
     check: bool,
     link_to_code: PathBuf,
+    dir: &str,
 ) -> anyhow::Result<()> {
-    format_sql(shell.clone(), check).await?;
-    rustfmt(shell.clone(), check, link_to_code).await?;
-    Ok(())
-}
-
-#[derive(Debug, Parser)]
-pub enum Formatter {
-    Rustfmt,
-    Contract,
-    Prettier {
-        #[arg(short, long)]
-        targets: Vec<Target>,
-    },
+    let mode = if check { "--check " } else { "" };
+    let full_path = link_to_code.join(dir).join("Cargo.toml");
+    let fmt_cmd = cmd!(shell, "cargo fmt --manifest-path {full_path} --all -- {mode}--config imports_granularity=Crate --config group_imports=StdExternalCrate");
+    Ok(Cmd::new(fmt_cmd).run()?)
 }
 
 #[derive(Debug, Parser)]
 pub struct FmtArgs {
     #[clap(long, short = 'c')]
     pub check: bool,
-    #[clap(subcommand)]
-    pub formatter: Option<Formatter>,
 }
 
 pub async fn run(shell: Shell, args: FmtArgs) -> anyhow::Result<()> {
     let ecosystem = EcosystemConfig::from_file(&shell)?;
     shell.set_var("ZKSYNC_USE_CUDA_STUBS", "true");
-    match args.formatter {
-        None => {
-            let mut tasks = vec![];
-            let extensions: Vec<_> = vec![Target::Js, Target::Ts, Target::Md, Target::Sol];
-            let spinner = Spinner::new(&msg_running_fmt_for_extensions_spinner(&extensions));
-            spinner.freeze();
-            for ext in extensions {
-                tasks.push(tokio::spawn(prettier(shell.clone(), ext, args.check)));
-            }
-            tasks.push(tokio::spawn(run_all_rust_formatters(
-                shell.clone(),
-                args.check,
-                ecosystem.link_to_code,
-            )));
-            tasks.push(tokio::spawn(prettier_contracts(shell.clone(), args.check)));
+    let spinner = Spinner::new(&msg_running_fmt_spinner());
+    let mut tasks = vec![];
+    tasks.push(tokio::spawn(prettier_contracts(shell.clone(), args.check)));
+    tasks.push(tokio::spawn(prettier(shell.clone(), args.check)));
+    tasks.push(tokio::spawn(format_sql(shell.clone(), args.check)));
+    for dir in ["core", "prover", "zkstack_cli"] {
+        tasks.push(tokio::spawn(rustfmt(
+            shell.clone(),
+            args.check,
+            ecosystem.link_to_code.clone(),
+            dir,
+        )));
+    }
```

Comment:
I'd love to keep an ability to format only one format and knowing what am i doing

Reply:
I've re-added sub-commands